File: JAVA/GraphsJava/BellmanFord.java
class Solution
{
    public int isNegativeWeightCycle(int V, int[][] graph)
    {
        //code here
        // Initialize distance of all vertices as infinite.
        int []dis = new int[V];
        for (int i=0; i<V; i++)
            dis[i] = Integer.MAX_VALUE;
     
        // initialize distance of source as 0
        dis[0] = 0;
        //Get number of edges
        int E = graph.length;
        // Relax all edges |V| - 1 times. 
        for (int i=0; i<V-1; i++)
        {
            for (int j = 0; j < E; j++)
            {
                if (dis[graph[j][0]] != Integer.MAX_VALUE && dis[graph[j][0]] + graph[j][2] < dis[graph[j][1]])
                    dis[graph[j][1]] = dis[graph[j][0]] + graph[j][2];
            }
        }
    int ans = 0;
    // check for negative-weight cycles.
    for (int i=0; i<E; i++)
    {
        int x = graph[i][0];
        int y = graph[i][1];
        int weight = graph[i][2];
        if (dis[x] != Integer.MAX_VALUE &&
                dis[x] + weight < dis[y])
            return 1;
    }
    return ans;
 
        
    }
}

File: C++/GraphsCPP/bellmanFord.cpp
Bellman Ford Algorith - GFG 

class Solution{
	public:
	/*  Function to implement Dijkstra
    *   adj: vector of vectors which represents the graph
    *   S: source vertex to start traversing graph with
    *   V: number of vertices
    */
    vector <int> bellman_ford(int V, vector<vector<int>> adj, int S) {
        // Code here
        vector<int> dis(V, 100000000);
        dis[S] = 0;
        
        // Relax for (V-1) times (Check Abdul Bari)
        for (int i = 0; i < V - 1; i++){
            
            // Check for all connected components
            for(int j = 0; j < adj.size(); j++) {
                int u = adj[j][0];
                int v = adj[j][1];
                int w = adj[j][2];
                
                if (dis[u]!=1e8 && dis[u]+w<dis[v])
                    dis[v] = dis[u] + w;
            }
        }
        
        return dis;
    }
};
